File: Bahasa/app/src/main/java/ds/id/Bahasa/Controls/SwipeViewPager.java
package ds.id.Bahasa.Controls;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import androidx.viewpager.widget.ViewPager;

public class SwipeViewPager extends ViewPager {
    private boolean enabled;

    public SwipeViewPager(Context context) {
        super(context);
        this.enabled = true;
    }

    public SwipeViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.enabled = true;
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        if (this.enabled) {
            return super.onInterceptTouchEvent(ev);
        }
        return false;
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        if (this.enabled) {
            return super.onTouchEvent(ev);
        }
        return false;
    }

    public void setPagingEnabled(boolean enabled) {
        this.enabled = enabled;
    }

}

File: Bahasa(kotlin)/app/src/main/java/ds/id/Bahasa/Controls/SwipeViewPager.kt
package ds.id.Bahasa.Controls

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager

class SwipeViewPager : ViewPager {

    private var enabled: Boolean? = false

    constructor(context: Context) : super(context) {
        enabled = true
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        enabled = true
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return if (enabled!!) {
            super.onInterceptTouchEvent(ev)
        } else false
    }

    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return if (enabled!!) {
            super.onTouchEvent(ev)
        } else false
    }

    fun setPagingEnabled(enabled: Boolean) {
        this.enabled = enabled
    }
}